baseline/pytorch/classify/model.py

```diff
@@ -197,33 +283,90 @@ def make_input(self, batch_dict):
+        * *hsz* -- (``List[int]``) The number of hidden units (defaults to 100)
         """
         kwargs.setdefault('hsz', [100])
         return super().init_stacked(input_dim, **kwargs)
 
 
 @register_model(task='classify', name='nbow')
 class NBowModel(NBowModelBase):
+    """Neural Bag-of-Words average pooling (standard) model"""
 
-    def init_pool(self, dsz, **kwargs):
-        return MeanPool1D(dsz)
+    def init_pool(self, input_dim: int, **kwargs) -> BaseLayer:
+        """Do average pooling on input embeddings, yielding a `dsz` output layer
+
+        :param input_dim: The word embedding depth
+        :param kwargs: None
+        :return: The average pooling representation
+        """
+        return MeanPool1D(input_dim)
 
 
 @register_model(task='classify', name='nbowmax')
 class NBowMaxModel(NBowModelBase):
+    """Max-pooling model for Neural Bag-of-Words.  Sometimes does better than avg pooling
+    """
 
-    def init_pool(self, dsz, **kwargs):
-        return MaxPool1D(dsz)
+    def init_pool(self, input_dim: int, **kwargs) -> BaseLayer:
+        """Do max pooling on input embeddings, yielding a `dsz` output layer
+
+        :param input_dim: The word embedding depth
+        :param kwargs: None
+        :return: The max pooling representation
+        """
+        return MaxPool1D(input_dim)
+
+
+@register_model(task='classify', name='fine-tune')
+class FineTuneModelClassifier(ClassifierModelBase):
+    """Fine-tune based on pre-pooled representations"""
+
+    def init_embed(self, embeddings: Dict[str, TensorDef], **kwargs) -> BaseLayer:
+        """This method creates the "embedding" layer of the inputs, with an optional reduction
+
+        :param embeddings: A dictionary of embeddings
+
+        :Keyword Arguments: See below
+        * *embeddings_reduction* (defaults to `concat`) An operator to perform on a stack of embeddings
+
+        :return: The output of the embedding stack followed by its reduction.  This will typically be an output
+          with an additional dimension which is the hidden representation of the input
+        """
+        return EmbeddingsStack(embeddings, reduction=kwargs.get('embeddings_reduction', 'concat'))
+
+    def init_stacked(self, input_dim: int, **kwargs) -> BaseLayer:
+        """Produce a stacking operation that will be used in the model
+
+        :param input_dim: The input dimension size
+        :param kwargs:
+        :return: A stacking operation (or None)
+        """
+        hszs = listify(kwargs.get('hsz', []))
+        if not hszs:
+            return PassThru(input_dim)
+        return DenseStack(input_dim, hszs, pdrop_value=self.pdrop)
+
+    def create_layers(self, embeddings: Dict[str, TensorDef], **kwargs):
+        self.embeddings = self.init_embed(embeddings, **kwargs)
+        self.stack_model = self.init_stacked(self.embeddings.output_dim, **kwargs)
+        self.output_layer = Dense(self.stack_model.output_dim, len(self.labels), activation="log_softmax")
```

Comment:
Is there a reason the FineTuneModel doesn't have a hook for overriding the output? It wouldn't be used much but it seems like it would be useful for something like multi-label classification

Reply:
good point, adding that now